### DISCUSSION TITLE
Legacy syntax highlights appear in the visual selection (only for the initial buffer load)

### DISCUSSION THREAD

**Discussion by creasty** [nvim-treesitter/nvim-treesitter#1539] [Q&A]
When I open a `.tsx` file and make a visual selection, highlights in the visual selection get rendered in different colors.
It can be fixed by reloading the buffer with `:e!` or opening another file with the same filetype, or by disabling the legacy syntax system completely with `:syntax off`.

Perhaps, some kind of race condition?

| 1. Initially opened | 2. Visual selection | 3. After `:e!` or  `:syntax off` |
|---|---|---|
| <img width="1003" alt="Screen Shot 2021-07-09 at 12 35 26" src="https://user-images.githubusercontent.com/1695538/125020238-757fa700-e0b3-11eb-9957-e3361d1509a8.png"> | <img width="1027" alt="Screen Shot 2021-07-09 at 12 35 51" src="https://user-images.githubusercontent.com/1695538/125019834-a4e1e400-e0b2-11eb-948b-77c0fee031f0.png"> | <img width="1096" alt="Screen Shot 2021-07-09 at 12 36 23" src="https://user-images.githubusercontent.com/1695538/125019838-a8756b00-e0b2-11eb-94e9-ec0d0f7f13f4.png"> | 

FYI, this is the same file rendered with the legacy syntax highlighter (`TSBufDisable highlight`):

<img width="300" alt="Screen Shot 2021-07-09 at 12 36 38" src="https://user-images.githubusercontent.com/1695538/125019843-a90e0180-e0b2-11eb-83db-d207e65ec552.png">

**Comment by vigoux** [comment]
Interesting, I'll convert that to an issue, but it seems that this could come from upstream.
@bfredl any idea on this?

**Reply by theHamsta** [reply]
This usually happens when you have some plugin installed that wants `syntax=on`

**Reply by creasty** [reply]
Looked over my config and plugins' code base, couldn't find any code that changes `syntax` option nor executes `:syntax` command (except for some in syntax definition files, which are certainly irrelevant to tsx).

c.f., https://github.com/creasty/dotfiles/tree/master/nvim

**Reply by vigoux** [reply]
To be sure, could you do `verbose set syntax?` ?
